core/server/data/migrations/versions/4.38/2022-03-02-09-10-update-free-products-visible-column.js

```diff
@@ -0,0 +1,64 @@
+const logging = require('@tryghost/logging');
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        const portalPlanSetting = await knex('settings').select('value').where('key', 'portal_plans').first();
+
+        if (!portalPlanSetting) {
+            logging.warn('Could not find portal_plans setting - skipping migration');
+            return;
+        }
+
+        try {
+            const settingData = JSON.parse(portalPlanSetting.value);
+
+            if (!settingData.includes('free')) {
+                logging.warn(`portal_plans does not include "free" - skipping migration`);
+                return;
+            }
+
+            logging.info(`Updating free products to visible`);
+            await knex('products').update('visible', true).where('type', 'free');
+        } catch (err) {
+            logging.warn('portal_plans setting is invalid - skipping migration');
+            return;
+        }
+    },
+    async function down(knex) {
+        const freeTier = await knex('products').select('id').where('type', 'free').first();
+        const portalPlanSetting = await knex('settings').select('value').where('key', 'portal_plans').first();
+
+        if (!freeTier) {
+            logging.info('Free tier is not visible, not updating portal_plans');
+            return;
+        }
+
+        if (!portalPlanSetting) {
+            logging.warn('Could not find portal_plans setting - skipping migration');
+            return;
+        }
+
+        try {
+            const existingSettingData = JSON.parse(portalPlanSetting.value);
+            let settingData;
+
+            if (freeTier.visible) {
+                if (existingSettingData.includes('free')) {
+                    logging.info('portal_plans setting already contains "free" - skipping update');
+                    return;
+                } else {
+                    settingData = existingSettingData.concat('free');
+                }
+            } else {
+                settingData = existingSettingData.filter(value => value !== 'free');
+            }
+
+            logging.info(`Updating portal_plans to ${settingData}`);
```

Comment:
Can we extract the `JSON.stringify(settingData)` out into a variable and use it in this log line? Without it, I get the following when rolling back:

```
[2022-03-02 20:58:13] INFO Updating portal_plans to
```

whereas I expect

```
[2022-03-02 20:58:13] INFO Updating portal_plans to []
```

Reply:
Done